ci/ray_ci/macos/macos_ci_test.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+set -ex
+
+export CI="true"
+export PYTHON="3.9"
+export RAY_USE_RANDOM_PORTS="1"
+export RAY_DEFAULT_BUILD="1"
+export LC_ALL="en_US.UTF-8"
+export LANG="en_US.UTF-8"
+export BUILD="1"
+export DL="1"
+
+
+bisect() {
+  # setup environment and build tools
+  MINIMAL_INSTALL=1 . ./ci/ci.sh init && source ~/.zshenv
+  source ~/.zshrc
+
+  # run bisect
+  test="$1"
+  passing_revision="$2"
+  failing_revision="$3"
+  bazel run //ci/ray_ci:bisect_test -- "$test" "$passing_revision" "$failing_revision"
+}
+
+
+run_single_test() {
+  # install dependencies
+  pip install -U --ignore-installed \
+    -c python/requirements_compiled.txt \
+    -r python/requirements.txt \
+    -r python/requirements/test-requirements.txt \
+    -r python/requirements/ml/dl-cpu-requirements.txt
```

Comment:
hmmm... maybe let me get my reef lambda test selection thing merged, so that you don't need to do this?

feels that this is repeating the code that is used to run macos tests? or why do you need this `run_single_test` function?

Reply:
the run_macos_tests is designed with a clean up mac environment in mind before and after running tests; which is not ideal for bisecting since we don't want to clean up the build artifacts in between bisects; let me think how i can just extract the logic of running tests and re-use more